File: week-04/day-02/GreenFox_org/Sponsor.py
from Person import Person

class Sponsor(Person):
    def __init__(self, company, hired_students):
        super().__init__()
        self.company = company
        self.hired_students = hired_students


    def get_goal(self):
        print("Hire brilliant junior software developers.")


    def introduce(self):
        print("Hi, I'm {}, a {} year old {}  who represents {} and hired {} students so far.".format(self.name, self.age, self.gender, self.company, self.hired_students))

    
    def hire(self, hired_students):
        hired_students += 1
    

File: Csharp-course/week-03/day-05/GreenFox/GreenFox/Sponsor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GreenFox
{
    class Sponsor : Person
    {
        protected string companyName;
        protected int hiredStudent;

        public Sponsor(string _name, int _age, string _gender, string company)
        {
            this.name = _name;
            this.age = _age;
            this.gender = _gender;
            this.companyName = company;
        }

        public Sponsor()
        {
            name = "JaneDoe";
            age = 30;
            gender = "female";
            this.companyName = "Google";
            this.hiredStudent = 0;
        }

        public override void Introduce()
        {
            Console.WriteLine("Hi, I'm {0}, a {1} year old {2} who represents {3} and hired {4} students so far.", this.name, this.age, this.gender, this.companyName, this.hiredStudent);
        }

        public void Hire()
        {
            this.hiredStudent++;
        }

        public override void GetGoal()
        {
            Console.WriteLine("Hire brilliant junior software developers!");
        }
    }
}
